### PR TITLE
Include Helm for checkov scans

### PR DESCRIPTION
**Pull request by andrewkreuzer**
# Proposed changes

Includes Helm for the running of checkov scans with the helm framework

## Readiness checklist

In order to have this pull request merged, complete the following tasks.

### Pull request author tasks

- [x] I included all the needed documentation for this change.
- [x] I provided the necessary tests.
- [x] I squashed all the commits into a single commit.
- [x] I followed the [Conventional Commit v1.0.0 spec](https://www.conventionalcommits.org/en/v1.0.0/).
- [x] I wrote the necessary upgrade instructions in the [upgrade guide](../docs/upgrade-guide.md).
- [x] If this pull request is about and existing issue,
  I added the `Fix #ISSUE_NUMBER` label to the description of the pull request.

### Super-linter maintainer tasks

- [x] Label as `breaking` if this change breaks compatibility with the previous released version.
- [x] Label as either: `automation`, `bug`, `documentation`, `enhancement`, `infrastructure`.

### CODE REVIEW

**Comment by ferrarimarco** [comment]
Hi @andrewkreuzer ! Thanks for this PR.

What problem are we solving here? :)

**Comment by andrewkreuzer** [comment]
when running checkov using the "helm" framework I wasn't seeing scans being done for my helm charts. Running locally I was able to get the desired output:
```
helm scan results:
   Passed checks: 233, Failed checks: 0, Skipped checks: 0
```
but wasn't seeing this output with superlinter

Running checkov in the superlinter container `checkov --framework helm --var-file my-values.yaml --directory charts` I was able to get to this error: 
```
2024-05-09 14:17:51,287 [MainThread  ] [ERROR]  There are no runners to run. This can happen if you specify a file type and a framework that are not compatible (e.g., `--file xyz.yaml --framework terraform`), or if you specify a framework with missing dependencies (e.g., helm or kustomize, which require those tools to be on your system). Running with LOG_LEVEL=DEBUG may provide more information.
```

**Comment by ferrarimarco** [comment]
I see, thanks for the details!

File: Dockerfile [hunk 1]
```diff
@@ -259,6 +259,13 @@ COPY dependencies/google-java-format /google-java-format
 RUN --mount=type=secret,id=GITHUB_TOKEN /install-google-java-format.sh \
     && rm -rfv /install-google-java-format.sh /google-java-format
 
+################
+# Install Helm #
+################
+COPY scripts/install-helm.sh /
+RUN /install-helm.sh \
+    && rm -rfv /install-helm.sh
```
Comment (ferrarimarco) [inline Dockerfile:267]: To have more controls over dependency updates, can you refactor this to grab helm from a container image?

Maybe this one https://hub.docker.com/r/alpine/helm

TLDR: Add a FROM statement near the others (see the top of this file), and copy the relevant binaries from that.

Also, that image appears to be a multi-platform image, so this will help when we'll eventually support arm64

File: scripts/install-helm.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
```
Comment (ferrarimarco) [inline scripts/install-helm.sh:1]: If we install helm from that container image, we don't need this script.

**Review by ferrarimarco — CHANGES_REQUESTED** [review]
Thanks for this PR!

Besides this, we would need some test cases (good and bad ones) in this dir: https://github.com/super-linter/super-linter/tree/main/test/linters/checkov

File: Dockerfile (hunk 1 above)
Comment (andrewkreuzer) [reply Dockerfile:267]: ya can do, I was trying to find an image built by the helm team but couldn't find one, I can use that one though!

**Comment by ferrarimarco** [comment]
One minor change: can you please make the test helm charts smaller? We don't need to have a fully working chart as a test case. What we need is the bare minimum to have tests pass :)

Thanks!

**Comment by andrewkreuzer** [comment]
@ferrarimarco reduced the size of the test charts hopefully that is small enough

**Comment by andrewkreuzer** [comment]
> @andrewkreuzer Thanks, the new test cases seems to be picked up. Nice!
> 
> Quick question: do the `good` test cases fail if you don't install helm? Just to confirm that we're covering this case as well.
> 
> Thanks!

no checkov will simple not run the helm checks if it can't find the helm binary

**Comment by ferrarimarco** [comment]
SGTM, I just found a reference about this in the [Checkov docs](https://www.checkov.io/7.Scan%20Examples/Helm.html)

**Review by ferrarimarco — APPROVED** [review]

File: Dockerfile [hunk 3]
```diff
@@ -259,6 +260,11 @@ COPY dependencies/google-java-format /google-java-format
 RUN --mount=type=secret,id=GITHUB_TOKEN /install-google-java-format.sh \
     && rm -rfv /install-google-java-format.sh /google-java-format
 
+################
+# Install Helm #
+################
+COPY --from=helm /usr/bin/helm /usr/bin/
```
Comment (ferrarimarco) [inline Dockerfile:266]: Can you add an entry for `helm` in the

- [linters list](https://github.com/super-linter/super-linter/blob/88ea3923a7e1f89dd485d079f6eb5f5e8f937589/test/inspec/super-linter/controls/super_linter.rb#L134), so we exercise it at least once? Please also add a comment explaining that this is not a linter, but we need it for Checkov.
- [files list](https://github.com/super-linter/super-linter/blob/88ea3923a7e1f89dd485d079f6eb5f5e8f937589/test/inspec/super-linter/controls/super_linter.rb#L459), so we check for its existence.

**Review by ferrarimarco — CHANGES_REQUESTED** [review]
Thanks a lot for all the work here.

We need just two more quick things about this in the test suite, and we should be good to go.

File: test/inspec/super-linter/controls/super_linter.rb [hunk 4]
```diff
@@ -160,6 +160,7 @@
     { linter_name: "goreleaser"},
     { linter_name: "google-java-format", version_command: "java -jar /usr/bin/google-java-format --version"},
     { linter_name: "hadolint"},
+    { linter_name: "helm"}, # not used as linter, needed for checkov's helm framework
```
Comment (ferrarimarco) [inline test/inspec/super-linter/controls/super_linter.rb:163]: Please, refactor this to `{ linter_name: "helm", version_option: "version"}` to fix the InSpec error.

Thanks!

**Review by ferrarimarco — CHANGES_REQUESTED** [review]

**Review by ferrarimarco — APPROVED** [review]
